File: db/src/main/java/org/delivery/db/user/UserRepository.java
package org.delivery.db.user;

import org.delivery.db.user.enums.UserStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserEntity, Long> {

    // select * from user where id = ? and status = ? order by id desc limit 1;
    Optional<UserEntity> findFirstByIdAndStatusOrderByIdDesc(Long id, UserStatus status);

    // select * from user where email =? and password = ? and status = ? order by id desc limit 1;
    Optional<UserEntity>findFirstByEmailAndPasswordAndStatusOrderByIdDesc(String email, String password, UserStatus status);
    // 얼티밋 버전은 쿼리 메소드 힌트도 줌
}

File: db/src/main/kotlin/org/delivery/db/user/UserRepository.kt
package org.delivery.db.user

import org.delivery.db.user.enums.UserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<UserEntity, Long> {
    // select * from user where id = ? and status = ? order by id desc limit 1;
    fun findFirstByIdAndStatusOrderByIdDesc(id: Long?, status: UserStatus?): UserEntity?

    // select * from user where email =? and password = ? and status = ? order by id desc limit 1;
    // 얼티밋 버전은 쿼리 메소드 힌트도 줌
    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): UserEntity?

}